### PR TITLE
fix: remove unconditional overwrite that disables teacher forcing in Seq2Seq

### LINKED ISSUE #1390: bug: Seq2Seq teacher forcing disabled by unconditional overwrite — intentional?
### Description

In `Seq2Seq.forward()` (`transliterate/thai2rom.py`), the teacher forcing if-else block is immediately overwritten by an unconditional assignment:

```python
# line 421-426
if teacher_force and target_seq is not None:
    decoder_input = target_seq[:, di].reshape(batch_size, 1)
else:
    decoder_input = topi.detach()

decoder_input = topi.detach()  # ← this always runs, overwriting the above
```

Line 426 makes the if-else block dead code — `teacher_forcing_ratio` has no effect.

### History

Looking at git history, this appears to be an incomplete revert:

1. **fd4b6676** (2019-08-02) — Original: teacher forcing with ternary expression. Correct.
2. **dd6410d4** (2022-10-24) — Removed teacher forcing, left `decoder_input = topi.detach()`.
3. **d48dc503** (2022-10-25) — "fix: bringback original thai2rom code" — added back the if-else but did not remove the old line from step 2.

### Question

Was removing teacher forcing in step 2 intentional? If the intent of step 3 was to restore it, then line 426 should be removed. The pre-trained model shipped with PyThaiNLP was trained before this change, so `romanize(engine="thai2rom")` still works — only retraining/fine-tuning is affected.

If teacher forcing is no longer needed (e.g. the model is only used for inference), the if-else block at lines 421-424 could be removed instead.

### Steps to reproduce

```python
import inspect
from pythainlp.transliterate.thai2rom import Seq2Seq

source = inspect.getsource(Seq2Seq.forward)
# Lines 421-426 show the if-else followed by unconditional overwrite
```

### PyThaiNLP version

5.3.3

### Files

- `pythainlp/transliterate/thai2rom.py` (lines 421-426)

### PR DESCRIPTION
**Pull request by phoneee**
### What do these changes do

Remove unconditional `decoder_input = topi.detach()` that overwrites the teacher forcing if-else block

Fixes #1390

- [x] Passed code styles and structures
- [x] Passed code linting checks and unit test

### CODE REVIEW

File: pythainlp/transliterate/thai2rom.py [hunk 1]
```diff
@@ -423,8 +423,6 @@ def forward(
             else:
                 decoder_input = topi.detach()
```
[the remark this answers is not in the document]
Comment (bact) [reply pythainlp/transliterate/thai2rom.py:424]: Note that since thai2rom uses torch, it's tests are in "noauto" test group.

The noauto group will not be run on the CI. Have to run the tests locally.
